File: SimTriggers/src/com/IconPippi/simtriggers/triggers/Trigger.java
package com.IconPippi.simtriggers.triggers;

import com.IconPippi.simtriggers.scripting.ScriptLoader;

public abstract class Trigger {
	
	protected final String method;
	
	protected final ScriptLoader scriptLoader = new ScriptLoader();
	
	protected Trigger(String method) {
		this.method = method;
	}
	
	protected abstract void trigger(); //Run the trigger
}

File: src/main/kotlin/com/simtriggers/fsx/triggers/Trigger.kt
package com.simtriggers.fsx.triggers

import com.simtriggers.fsx.event.RegisterEvent
import com.simtriggers.fsx.scripting.ScriptLoader

/**
 * 04/08/2019
 * This abstract class explains itself
 *
 * @author IconPippi
 */
abstract class Trigger(private val function: String, private val eventName: String?) {

    /** Event registerer */
    protected val eventRegister: RegisterEvent = RegisterEvent()

    /** Script loader */
    protected val scriptLoader: ScriptLoader = ScriptLoader()

    /**
     * Run the trigger
     * @param eventName Trigger's event
     */
    abstract fun trigger(eventName: String?)

}